include/openssl/dh.h

```diff
@@ -136,9 +136,9 @@ DECLARE_ASN1_ITEM(DHparams)
         ASN1_d2i_bio_of(DH, DH_new, d2i_DHxparams, bp, x)
 #   define i2d_DHxparams_bio(bp, x) \
         ASN1_i2d_bio_of(DH, i2d_DHxparams, bp, x)
-#  endif
 
-DECLARE_ASN1_DUP_FUNCTION_name(DH, DHparams)
+DECLARE_ASN1_DUP_FUNCTION_name_attr(OSSL_DEPRECATEDIN_3_0, DH, DHparams)
+#  endif
 
 DEPRECATEDIN_3_0(const DH_METHOD *DH_OpenSSL(void))
```

Comment:
I'd like to suggest that you do like I did in #13096, and modify all the deprecations in this file to follow https://github.com/openssl/openssl/blob/master/doc/internal/man7/deprecation.pod ?

Reply:
Done.